src/mon/FSCommands.cc

```diff
@@ -384,6 +384,17 @@ class SetHandler : public FileSystemCommandHandler
       return -EINVAL;
     }
 
+  bool confirm = false;
+  cmd_getval(cmdmap, "yes_i_really_mean_it", confirm);
+  if (var == "max_mds" && !confirm && mon->mdsmon()->has_health_warnings()) {
+    ss << "One or more file system health warnings are present. Modifying "
```

Comment:
nit: One or more file systems have health warnings. Modifying "max_mds" may destabilize the system further. Use `--yes...` to override.

Reply:
I had something similar, current version was recommended by Patrick.